### PR TITLE
chore: automatically apply labels to PRs based on files modified

### LINKED ISSUE #62: Automatically apply technology/sgx label
Get a list of the files modified. Do a regular expression search for SGX. If found, label it with `technology/sgx`.

### PR DESCRIPTION
**Pull request by platten**
Signed-off by Paul Pietkiewicz [paul@profian.com](mailto:paul@profian.com)

Rebased version of @shravi24's PR #1066
Closes #62

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+- helper/**/*
+
+amd sev:
+- internal/shim-sev/**/*
+
+intel sgx:
+- internal/shim-sgx/**/*
+
+wasm:
+- internal/wasmldr/**/*
```
Comment (haraldh) [inline .github/labeler.yml:12]:
```suggestion
- src/bin/wasmldr/**/*
```

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- tests/wasm/**/*
```
Comment (haraldh) [inline .github/labeler.yml:13]:
```suggestion
- tests/wasm/**/*
- tests/wasmldr_tests.rs
```

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+infrastructure:
+- .github/**/*
+- helper/**/*
+
+amd sev:
+- internal/shim-sev/**/*
```
Comment (haraldh) [inline .github/labeler.yml:6]:
```suggestion
- internal/shim-kvm/**/*
```

**Review by haraldh — CHANGES_REQUESTED** [review]

File: .github/labeler.yml (hunk 3 above)
Comment (rvolosatovs) [reply .github/labeler.yml:6]: Shouldn't this also include the backend?

Also change at this path should probably label both as SEV and KVM

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+intel sgx:
+- internal/shim-sgx/**/*
```
[the remark this answers is not in the document]
Comment (haraldh) [reply .github/labeler.yml:9]: yes

[the remark this answers is not in the document]
Comment (platten) [reply .github/labeler.yml:18]: Please let me know if this is what you had in mind.

[the remark this answers is not in the document]
Comment (haraldh) [reply .github/labeler.yml:18]: no...

```suggestion
- internal/shim-sgx/**/*
- src/backend/sgx/**/*
```

File: .github/labeler.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+- helper/**/*
+
+backend:
+- internal/shim-kvm/**/*
+- internal/shim-sgx/**/*
+- src/bin/wasmldr/**/*
+- src/**/*
+
+kvm:
+- internal/shim-kvm/**/*
```
Comment (haraldh) [inline .github/labeler.yml:12]:
```suggestion
kvm:
- internal/shim-kvm/**/*
- src/backend/kvm/**/*
```

File: .github/labeler.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+amd sev:
+- internal/shim-kvm/**/*
```
Comment (haraldh) [inline .github/labeler.yml:15]:
```suggestion
amd sev:
- internal/shim-kvm/**/*
- src/backend/sev/**/*
```

File: .github/labeler.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+infrastructure:
+- .github/**/*
+- helper/**/*
+
+backend:
+- internal/shim-kvm/**/*
+- internal/shim-sgx/**/*
+- src/bin/wasmldr/**/*
+- src/**/*
```
Comment (haraldh) [inline .github/labeler.yml:9]:
```suggestion
backend:
- src/backend/**/*
```

**Review by haraldh — CHANGES_REQUESTED** [review]

**Review by haraldh — APPROVED** [review]

**Pull request merged**